provider/cmd/manifest.go

```diff
@@ -57,15 +66,75 @@ func doSendManifest(cmd *cobra.Command, sdlpath string) error {
 		return err
 	}
 
-	cert, err := cutils.LoadAndQueryCertificateForAccount(cmd.Context(), cctx, cctx.Keyring)
+	// owner address in FlagFrom has already been validated thus save to just pull its value as string
+	providers, err := providersFromFlagsOrQuery(cmd.Flags(), cmd.Context(), cctx, dtypes.DeploymentID{
+		Owner: cctx.GetFromAddress().String(),
+		DSeq:  dseq,
+	})
+	if err != nil {
+		return err
+	}
+
+	type result struct {
+		Provider sdk.Address `json:"provider" yaml:"provider"`
+		Status   string      `json:"status" yaml:"status"`
+		Error    error       `json:"error,omitempty" yaml:"error,omitempty"`
+	}
+
+	results := make([]result, 0, len(providers))
+
+	submitFailed := false
+
+	for _, provider := range providers {
+		gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), provider, []tls.Certificate{cert})
+		if err != nil {
+			return err
+		}
+
+		err = gclient.SubmitManifest(context.Background(), dseq, mani)
+		res := result{
+			Provider: provider,
+			Status:   "PASS",
+			Error:    err,
+		}
+
+		if err != nil {
+			res.Status = "FAIL"
+			submitFailed = true
+		}
+
+		results = append(results, res)
+	}
+
+	buf := &bytes.Buffer{}
+
+	switch cmd.Flag("output").Value.String() {
+	case outputText:
+		for _, res := range results {
+			_, _ = fmt.Fprintf(buf, "provider: %s\n\tstatus: %s\n", res.Provider, res.Status)
+			if res.Error != nil {
+				_, _ = fmt.Fprintf(buf, "\terror: %v\n", res.Error)
+			}
+		}
+	case outputJson:
+		err = json.NewEncoder(buf).Encode(results)
+	case outputYaml:
+		err = yaml.NewEncoder(buf).Encode(results)
+	}
+
 	if err != nil {
 		return err
 	}
 
-	gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), prov, []tls.Certificate{cert})
+	_, err = fmt.Fprint(cmd.OutOrStdout(), buf.String())
+
 	if err != nil {
 		return err
 	}
 
-	return showErrorToUser(gclient.SubmitManifest(context.Background(), dseq, mani))
+	if submitFailed {
+		return errors.Errorf("submit manifest to some providers has been failed")
```

Comment:
This can be assigned statically I think like `var errSubmitManifestFailed = errors.New(...)`